guides/source/association_basics.md

```diff
@@ -355,2288 +447,2474 @@ class CreateAccountHistories < ActiveRecord::Migration[7.2]
+deletion affects only the join records, not the associated records.
+
+When using `dependent: :destroy` on a scoped association, only the scoped
+objects are destroyed. For example, in a `Post` model defined as `has_many
+:comments, -> { where published: true }, dependent: :destroy`, calling destroy
+on a post will only delete published comments, leaving unpublished comments
+intact with a foreign key pointing to the deleted post.
+
+You cannot use the `:dependent` option directly on a `has_and_belongs_to_many`
+association. To manage deletions of join table records, handle them manually or
+switch to a `has_many :through` association, which provides more flexibility and
+supports the `:dependent` option.
+
+#### `:foreign_key`
+
+By convention, Rails assumes that the column used to hold the foreign key on
+this model is the name of the association with the suffix `_id` added. The
+`:foreign_key` option lets you set the name of the foreign key directly:
 
 ```ruby
-@assembly = @part.assemblies.build({ assembly_name: "Transmission housing" })
+class Supplier < ApplicationRecord
+  has_one :account, foreign_key: "supp_id"
+end
 ```
 
-##### `collection.create(attributes = {})`
+NOTE: Rails does not create foreign key columns for you. You need to explicitly
+define them in your migrations.
+
+#### `:primary_key`
 
-The [`collection.create`][] method returns a new object of the associated type. This object will be instantiated from the passed attributes, the link through the join table will be created, and, once it passes all of the validations specified on the associated model, the associated object _will_ be saved.
+By default, Rails uses the `id` column as the primary key for its tables. The
+`:primary_key` option allows you to specify a different column as the primary
+key.
+
+For example, if the `users` table uses `guid` as the primary key instead of
+`id`, and you want the `todos` table to reference `guid` as a foreign key
+(`user_id`), you can configure it like this:
 
 ```ruby
-@assembly = @part.assemblies.create({ assembly_name: "Transmission housing" })
+class User < ApplicationRecord
+  self.primary_key = 'guid' # Sets the primary key to guid instead of id
+end
+
+class Todo < ApplicationRecord
+  belongs_to :user, primary_key: 'guid' # References the guid column in users table
+end
 ```
 
-##### `collection.create!(attributes = {})`
+When you execute `@user.todos.create`, the `@todo` record will have its
+`user_id` value set to the `guid` value of `@user`.
 
-Does the same as `collection.create`, but raises `ActiveRecord::RecordInvalid` if the record is invalid.
+`has_and_belongs_to_many` does not support the `:primary_key` option. For this
+type of association, you can achieve similar functionality by using a join table
+with has_many `:through` association, which gives more flexibility and supports
+the `:primary_key` option. You can read more about this in the [`has_many
+:through` section](#has-many-through).
```

Comment:
This is formatted a bit strangely. Not sure how to fix it:
<img width="829" alt="image" src="https://github.com/user-attachments/assets/1f420071-bb7e-4301-a644-a1a42f407610">

Reply:
Fixed 3 instances ✅
<img width="855" alt="Screenshot 2024-09-01 at 22 41 56" src="https://github.com/user-attachments/assets/8cda4944-43ff-40e8-b52f-c4634fd88d78">

Reply:
Nice!